CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.13)
+cmake_minimum_required(VERSION 3.24)
```

Comment:
Why do we need to upgrade here?

Reply:
CMAKE_COMPILE_WARNING_AS_ERROR was new in 3.24 . I don't think its really necessary, but it's what I was asked to do when I did this for xeus-cpp.

Reply:
I do not think we should upgrade cmake to that late versions. That'd leave behind a lot of clients.

Reply:
It is probably fine as xeus-cpp is a end-point project not a library that needs to be incorporated in many contexts...